src/Base.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+class Base {
+  constructor (name, code, implementation, alphabet) {
```

Comment:
- **implementation** defines the module in which the actual encoding/decoding functions are implemented. see [this](https://github.com/multiformats/js-multibase/pull/7/files#diff-a7748843fbda5167d065231e3a2e7a1fR8) in constants.js for the use case. An implementation is a class taking an alphabet as an argument because it's a key factor in the encoding/decoding process (following base-x architecture).

- The idea behind this is that several bases may have implementations from different locations/modules so it's useful to have an object (and a summary) of all of them in one location (hence the constants.js). But I'm open to any refactoring for this (although I believe the multi-sourced implementations statement should be taken into account). Also even if the unary base is deployed in the base-x module (it is missing in base-x right now, which explains the empty string in the base1 list of arguments in constants.js), **given the very nature of the base-x encoding/decoding algorithm in base-x his author seems to imply that there is no possible support for padded encodings** (and indeed, after a code lookup it seems to be only bitwise manipulation on the input yielding the new base representation.